src/App.js

```diff
@@ -28,14 +28,16 @@ const App = () => {
       <div className="App">
         <Switch>
           <Route exact path="/">
+            <Home />
+          </Route>
+          <Route exact path="/list">
             <List />
           </Route>
 
           <Route exact path="/add-item">
             <AddItem />
           </Route>
         </Switch>
-        <NavBar />
 
```

Comment:
![navbar-conditional-suggestion](https://user-images.githubusercontent.com/47455758/114732411-1aceed00-9d08-11eb-9989-97321fe052c5.png)

Great job getting those tokens working! I pulled it down, and it looks and works awesome! 

My only suggestion is to move the NavBar component back to App.js inside the `<Router>`. 

(For the sake of the screen shot, I removed the `<div>` holding the rendered Grocery List).

When you move the NavBar out of the switch, it will persist on the screen, which we don't want when the user is on the Home screen and has not created a token yet.

To solve this, I imported the `checkForToken` function from `./lib/localStorage` and just like inside `Home.js` created the `isThereToken` variable: `const isThereToken = checkForToken('token', '');`

Finally I created a ternary just underneath the closing `</Switch>` that looks like `{isThereToken ? <NavBar /> : null}` so that when there is no token, the NavBar won't show up on the home screen. 

And the final part was removing `<NavBar />` from both `List.js` and `AddItem.js`.

Reply:
I do like the idea of leaving the `<NavBar />` component in one place. It's not a huge deal here since there's only two views, but it does better align with DRY code and may be a little more flexible for future features. Thinking ahead, if you'd like to leave some version of the NavBar there on the home page/other pages (ex: a solid bar), you could move the ternary into the `<NavBar />` and only render those menu options when there's a token.

Reply:
Thank you so much @jamesncox and @laurenyz ! Bringing the `<NavBar>` component into both views was bothering me as well. I can (usually) get things to work, but there's always the feeling that there was an easier way to go about it 😅 I'll definitely take your suggestion!

Reply:
Hey! We tried implementing the suggestion, and we ran into another issue. The ternary condition is only evaluated once when `App.js` is rendered upon initial page load. Therefore, when we switch routes, it isn't executed and we end up with no `NavBar` when we want it. 

We were trying to solve this but haven't come up with any solutions yet. It would be great if anyone would like to share any insight or has input regarding this blip!

https://user-images.githubusercontent.com/38255956/114934325-62847000-9dff-11eb-8b7c-e0def65938f2.mov

Reply:
I see that! I just replicated the work I suggested locally, and I'm noticing the same bug. I honestly don't recall seeing that bug yesterday! 

I wonder, does React Router stop the page from updating if something changes?

Reply:
Ah ok, I think what's happening is that the function is only running when the app is initially loaded. Since app is never unmounting or re-rendering with the the router changes. 

Initial thoughts:

1. You could use a [useEffect](https://reactjs.org/docs/hooks-effect.html) that only checks for a token on load of the page. You can store the token with [useState](https://reactjs.org/docs/hooks-state.html). Then, as part of the `sendItem` function, you can update the token as well.
2. You could use a useEffect and have a it update on route change (you'd need to add something in the dependency array), but I think that would result in unnecessary useEffects firing off.

Reply:
@jssckbl @apdsrocha I made a new branch instead of using making the code suggestions here because it affected a few files. Feel free not to look at it if you would prefer to experiment on your own! The branch name is [ly-navbar-token-suggestion](https://github.com/the-collab-lab/tcl-23-smart-shopping-list/commit/c270e4c8ae7831a497d83eba5e81bcca4498752d)

Reply:
I agree with @laurenyz suggestion for 1. You won't need to useEffect to update on route change. You can use the token state to determine what route to display with `<Redirect>`.

Reply:
I also support the use of `useEffect` here. It's a great way to plug into the lifecycle of your component, instructing it to re-render when state changes, in this case the state of the token! You've got all the foundational bits to make this render super nicely. 👏